hack/verify-generate.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+#####
+#  Used to ensure that running `make generate` results in a non-dirty repository.
+#  Used to verify that make generate wasn't forgotten for changes that may require generation.
+#####
+
+set -o nounset
+set -o pipefail
+# intentionally not setting 'set -o errexit' because we want to print custom error messages
+
+# versions of tools
+echo "controller-gen $(controller-gen --version)"
+
+# make sure make generate can be invoked
+make generate
+RETVAL=$?
+if [[ ${RETVAL} != 0 ]]; then
+    echo "Invoking 'make generate' ends with non-zero exit code."
+    exit 1
+fi
+
+git diff --exit-code --quiet
+RETVAL=$?
+
+if [[ ${RETVAL} != 0 ]]; then
+    echo "Running 'make generate' produces changes to the current git status. Maybe you forgot to check-in your updated generated files?"
+    echo "The current diff: $(git diff)"
```

Comment:
```suggestion
    echo "The current diff:"
    git diff
```
No need to make the shell load all that output and pass it as a giant argument to echo. Might go bad if it's large.